### PR TITLE
Update json-editor extension

### PR DESCRIPTION
**Pull request by jvrbnjak89**
## Description

- Support passing options to [json-ts](https://github.com/shakyshane/json-ts?tab=readme-ov-file#options) when converting result to typescript definition
- Make options configurable from extension preferences

## Screencast

Screenshot 2024-07-30 at 11 46 25

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/json-editor/package.json [hunk 1]
```diff
@@ -18,6 +21,40 @@
...
+      "type": "textfield",
+      "required": false,
+      "default": "I",
+      "title": "Typescript Definition - Prefix"
+    },
+    {
+      "name": "rootName",
+      "description": "Override the RootObject name of the top-level interface",
+      "type": "textfield",
+      "required": true,
```
Comment (pernielsentikaer) [inline extensions/json-editor/package.json:53]: Can we make this also not required?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Can you give me some examples I can test with?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/json-editor/package.json (hunk 1 above)
Comment (jvrbnjak89) [reply extensions/json-editor/package.json:53]: Hi @pernielsentikaer, you can test with any valid json and I personally used escaped one like this:

```
{\"id\":\"9daba9b2-7de5-4153-ba3e-2a867d29f411\",\"created_at\":\"2023-03-25T09:51:38.466819+00:00\",\"updated_at\":\"2023-03-26T16:24:53.067016+00:00\",\"deleted_at\":null,\"group_id\":\"0092d2c2-1a90-4894-836e-d61d04518038\"}
```

Which gives a prettified json you can convert to typescript definition using copy to clipboard action.
Screenshot 2024-07-30 at 14 18 47

The result is the following interface:
code

Comment (jvrbnjak89) [reply extensions/json-editor/package.json:53]: Regarding the required `rootName` I noticed an issue when it's empty/undefined even though it's allowed by the type definition. However, now that I'm thinking of it again, I guess I can set a default value from the code, not the preferences list.

I'll get back to you on that one and thank you for the suggestion :)

**Comment by pernielsentikaer** [comment]
Thanks for the update @jvrbnjak89 🔥 

@lryong could you check this?

**Review by pernielsentikaer — APPROVED** [review]

**Pull request merged**